src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -947,32 +950,37 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
 
         // wait for automation to appear in entity registry when creating a new automation
         if (entityRegPromise) {
-          const timeout = setTimeout(() => {
-            this._readOnly = true;
-            this._dirty = false;
-            showAlertDialog(this, {
-              title: this.hass.localize(
-                "ui.panel.config.automation.editor.new_automation_setup_failed_title",
-                {
-                  type: this.hass.localize(
-                    "ui.panel.config.automation.editor.type_automation"
-                  ),
-                }
-              ),
-              text: this.hass.localize(
-                "ui.panel.config.automation.editor.new_automation_setup_failed_text",
-                {
-                  type: this.hass.localize(
-                    "ui.panel.config.automation.editor.type_automation"
-                  ),
-                }
-              ),
-              warning: true,
-            });
-          }, 5000);
-          const automation = await entityRegPromise;
-          clearTimeout(timeout);
-          entityId = automation.entity_id;
+          try {
+            const automation = await promiseTimeout(2000, entityRegPromise);
+            entityId = automation.entity_id;
+          } catch (e) {
+            if (e instanceof Error && e.name === "TimeoutError") {
+              this._readOnly = true;
+              this._saveFailed = true;
```

Comment:
Shouldn't we instead of this, just navigate to the new automation and let that handle it?

```
        navigate(`/config/automation/edit/${id}`, { replace: true });
```

Reply:
If we let user navigate through to the other side, what should the behavior be? Just allow continue editing as normal? I guess editing can continue any registry based settings will be broken.

Was kind of thinking best to just kick them out ASAP and go fix the config before resuming.

Reply:
I would still show the alert message, but then just navigate?

It will be the same when they refresh the page or go back and visit any other automation right?

Reply:
So if I show the dialog and do nothing else but continue, the editor can continue, it will have no entity id, so most of the overflow menu items are disabled. If you rename the automation and again select an area it will show the dialog again. 

If that sounds good I'll go for that.

Reply:
Sounds good